src/main/java/net/sf/jabref/exporter/SaveActions.java

```diff
@@ -7,41 +7,91 @@
 import net.sf.jabref.logic.formatter.IdentityFormatter;
 import net.sf.jabref.model.entry.BibEntry;
 
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
+import java.util.*;
 
 public class SaveActions {
 
-    private final Map<String, Formatter> actions;
+    private final List<SaveAction> actions;
 
     private List<Formatter> availableFormatters;
 
     public static final String META_KEY = "saveActions";
 
+    private boolean enabled;
+
     public SaveActions(MetaData metaData) {
-        actions = new HashMap<>();
+        if (metaData == null) {
+            throw new IllegalArgumentException("MetaData must not be null");
+        }
+        actions = new ArrayList<>();
         setAvailableFormatters();
 
         List<String> formatters = metaData.getData(META_KEY);
-        if(formatters==null) {
+        if (formatters == null) {
             // no save actions defined in the meta data
             return;
         } else {
-            for (int i = 0; i < formatters.size(); i += 2) {
-                try {
-                    String field = formatters.get(i);
-                    Formatter formatter = getFormatterFromString(formatters.get(i + 1));
-
-                    actions.put(field, formatter);
-                } catch (IndexOutOfBoundsException e) {
-                    // the meta data string in the file is broken. -> Ignore the last item
-                    break;
-                }
+            parseEnabledStatus(formatters);
+
+            parseSaveActions(formatters);
+        }
+
+    }
+
+    public boolean isEnabled() {
+        return enabled;
+    }
+
+    public List<SaveAction> getConfiguredActions() {
+        return Collections.unmodifiableList(actions);
+    }
+
+    public List<Formatter> getAvailableFormatters() {
+        return Collections.unmodifiableList(availableFormatters);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        SaveActions that = (SaveActions) o;
+
+        if (enabled != that.enabled) return false;
+        return actions.equals(that.actions);
+
+    }
+
+    @Override
+    public int hashCode() {
+        int result = actions.hashCode();
+        result = 31 * result + (enabled ? 1 : 0);
+        return result;
+    }
+
+    private void parseSaveActions(List<String> formatters) {
+        //read concrete actions
+        for (int i = 1; i < formatters.size(); i += 2) {
```

Comment:
Yes, it would make sense to get that right the first time :) I am open to improvements regarding the format. Currently, it looks like this 
```
@comment{jabref-meta: saveActions:enabled;title;LowerCaseChanger;}
```

Do you have a suggestion for a better format?